source/falaise/snemo/datamodels/sim_calo_digi_hit.h

```diff
@@ -17,32 +17,58 @@
 
 namespace snemo {
 
-namespace datamodel {
+	namespace datamodel {
 
-/// \brief Simulated calo digitized hit.
-class sim_calo_digi_hit : public mctools::digitization::sampled_signal {
- public:
-  /// Constructor
-  sim_calo_digi_hit();
+		/// \brief Simulated calo digitized hit.
+		class sim_calo_digi_hit : public mctools::digitization::sampled_signal
```

Comment:
Given the simplicity of this data (I'm going from `RDD/SDD` in DocDB-4612), could the inheritance be replaced with composition here, e.g.

```c++
class sim_calo_digi_hit {
  ...
  private:
    geomtools::geom_id geomID_;
    geomtools::geom_id elecID_;
    ...
    std::vector<int16_t> waveform; // maybe even std::array if size known
    ... sample frequency etc?
};
```

Or, if attributes go along with the waveform, then define an additional struct to organise this? The `mctools` object brings a lot of complexity the requirement for which is not obvious yet. Composition can also make persistency easier, and schema evolution more robust.